qiskit/algorithms/evolvers/real/trotterization/trotter_qrte.py

```diff
@@ -0,0 +1,178 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An algorithm to implement a Trotterization real time-evolution."""
+
+from typing import Union, Dict, Optional
+
+from qiskit.algorithms import EvolutionProblem, EvolutionResult, RealEvolver, eval_observables
+from qiskit.circuit import Parameter
+from qiskit.opflow import (
+    OperatorBase,
+    SummedOp,
+    PauliOp,
+    CircuitOp,
+    ExpectationBase,
+    CircuitSampler,
+)
+from qiskit.circuit.library import PauliEvolutionGate
+from qiskit.providers import Backend, BaseBackend
+from qiskit.synthesis import ProductFormula, LieTrotter
+from qiskit.utils import QuantumInstance, algorithm_globals
+from .trotter_ops_validator import is_op_bound
+
+
+class TrotterQrte(RealEvolver):
+    """ Class for performing Quantum Real Time Evolution using Trotterization.
+    Type of Trotterization is defined by a ProductFormula provided.
+
+    Examples:
+
+        .. jupyter-execute::
+
+            from qiskit.opflow import X, Y, Zero
+            from qiskit.algorithms import EvolutionProblem, EvolutionResult
+            from qiskit.algorithms.evolvers.real.implementations.\
+                trotterization.trotter_qrte import TrotterQrte
+
+            operator = X + Z
+            initial_state = Zero
+            time = 1
+            evolution_problem = EvolutionProblem(operator, 1, initial_state)
+            # LieTrotter with 1 rep
+            trotter_qrte = TrotterQrte(evolution_problem)
+            evolved_state = trotter_qrte.evolve().evolved_state
+    """
+
+    def __init__(
+        self,
+        quantum_instance: Union[QuantumInstance, BaseBackend, Backend],
+        product_formula: ProductFormula = LieTrotter(),
```

Comment:
Having a mutable object as a default value can lead to unexpected behavior since Python creates a single global instance for this default which is reused. This is why you will often see default of None and if None internally it creates a new default instance each time.

Reply:
Very good point, thanks.